### PR TITLE
Extra usb reports

### LINKED ISSUE #221: Include extra USB reports for shortcuts for maximum compatibility
Currently, UHK shortcuts are sent via USB using the smallest number of USB reports possible. For example, in the case of Ctrl+C, both the Ctrl modifier and the scancode of C gets sent in the same report at once. This is correct, but it messes up some misbehaving applications like Karabiner and Hyper-V. By holding and releasing modifiers in extra reports, this might be worked around at the firmware level:

* Report 1: Hold Ctrl
* Report 2: Hold Ctrl + send the scancode of C
* Report 3: Hold Ctrl

Hopefully, this will work out fine.

### PR DESCRIPTION
**Pull request by kareltucek**
Closes #221

### CODE REVIEW

File: right/src/usb_report_updater.c [hunk 1]
```diff
@@ -274,25 +274,27 @@ static void applyKeyAction(key_state_t *keyState, key_action_t *action, uint8_t
                 } else {
                     ActiveUsbBasicKeyboardReport->modifiers |= action->keystroke.modifiers;
                 }
-                switch (action->keystroke.keystrokeType) {
-                    case KeystrokeType_Basic:
-                        if (basicScancodeIndex >= USB_BASIC_KEYBOARD_MAX_KEYS || action->keystroke.scancode == 0) {
+                if(action->keystroke.modifiers == 0 || KeyState_ActivatedEarlier(keyState)) {
```
Comment (kareltucek) [inline right/src/usb_report_updater.c]: Just this line actually changes in this block.

**Comment by kareltucek** [comment]
@mondalaci I suspect that there might be some unpublished comments too?

File: right/src/macros.c [hunk 2]
```diff
@@ -348,35 +367,74 @@ bool processScrollMouseAction(void)
...
+
 bool processTextAction(void)
 {
     static uint16_t textIndex;
     static uint8_t reportIndex = USB_BASIC_KEYBOARD_MAX_KEYS;
     char character;
     uint8_t scancode;
+    uint8_t mods;
 
+    /* When all characters have been sent, finish. Yet, one last report should still be sent
```
Comment (mondalaci) [inline right/src/macros.c]: Please change every comment to:

`// My comment ...`

style.

File: right/src/usb_report_updater.c [hunk 3]
```diff
@@ -311,34 +311,40 @@ static void applyKeyAction(key_state_t *keyState, key_action_t *action, uint8_t
 
         switch (action->type) {
             case KeyActionType_Keystroke:
+            {
+                bool stickyModifiersUnChanged = true;
```
Comment (mondalaci) [inline right/src/usb_report_updater.c]: It'd be easy to comprehend a non-negated variable like `stickyModifiersChanged`.

Comment (kareltucek) [reply right/src/usb_report_updater.c]: Sure, however result is having to negate its every use further down the road so I am not sure if it is much of an improvement...

**Pull request merged**